SDKAndroid/src/main/java/app/cybrid/sdkandroid/flow/TradeFlow.kt

```diff
@@ -108,317 +102,442 @@ class TradeFlow @JvmOverloads constructor(
+                            fontWeight = FontWeight.Bold,
                             fontSize = 16.sp
-                        ),
-                        singleLine = true,
-                        colors = TextFieldDefaults.textFieldColors(
-                            textColor = Color.Black,
-                            cursorColor = colorResource(id = R.color.primary_color),
-                            backgroundColor = Color.Transparent,
-                            focusedIndicatorColor = colorResource(id = R.color.list_prices_asset_component_code_color),
-                            unfocusedIndicatorColor = colorResource(id = R.color.list_prices_asset_component_code_color),
-                            disabledIndicatorColor = colorResource(id = R.color.list_prices_asset_component_code_color)
                         )
-                    )
-                    Image(
-                        painter = painterResource(id = R.drawable.ic_change),
-                        contentDescription = "Change icon to crypto-fiat",
-                        contentScale = ContentScale.Crop,
-                        modifier = Modifier
-                            .padding(top = 18.dp)
-                            .padding(horizontal = 16.dp)
-                            .width(16.dp)
-                            .height(21.dp)
-                            .align(Alignment.End)
-                            .clickable {
-                                if (currentValueInput.value == AssetBankModel.Type.fiat) {
-                                    currentValueInput.value = AssetBankModel.Type.crypto
-                                } else {
-                                    currentValueInput.value = AssetBankModel.Type.fiat
-                                }
-                            }
-                    )
+                    }
+                )
+            }
+        }
+    }
 
-                    // -- Spinner
-                    OutlinedTextField(
-                        value = currencyState.value.name,
-                        onValueChange = {},
-                        interactionSource = remember { MutableInteractionSource() }
-                            .also { interactionSource ->
-                                LaunchedEffect(interactionSource) {
-                                    interactionSource.interactions.collect { iteration ->
-                                        if (iteration is PressInteraction.Release) {
-                                            expanded.value = !expanded.value
-                                        }
-                                    }
-                                }
-                            },
-                        label = {
-                            Text(
-                                text = "Currency"
-                            )
-                        },
+    @Composable
+    private fun CryptoCurrencyInput(
```

Comment:
We need to support the input of either the asset or the counterasset. We should stay away from terms like crypto and fiat if we can and stick with asset and counter asset. It's more generic and will avoid special casing.

Reply:
Chenged to: PreQuoteCurrencyInput